### ISSUE TITLE
Dependency Management Part I

### ISSUE THREAD

**Issue by aorinevo** [NerdWalletOSS/shepherd#103]
## Introduction

Shepherd has a number of advisory exceptions when installing off of v1.4.1 ranging from low to high.
```
(544 low, 14 moderate, 85 high)
```

## Proposed Changes

Update dependencies to reduce vulnerabilites

### Phase I
- Packages
  - simple-git
  - ora
  - @types/ora
  - @types/joi -> @types/hapi__joi
  - joi -> @hapi/joi
- CI changes
  - drop support for Node 6; support Node 8, 10, and 12

### Phase II
- chalk
- child-process-promise (already on latest)
- js-yaml
- lodash
- log-symbols 

### Phase III
- Packages
  - commander
  - fs-extra
  - fs-extra-promise
  - netrc
- Audit
  - run `npm audit fix` 

## Notes
- Breaking out remaining deps into a separate issue as those will require large changes.

## Screenshots

<img width="657" alt="Screen Shot 2020-04-05 at 8 39 45 PM" src="https://user-images.githubusercontent.com/3240137/78514228-4b3b8d00-777e-11ea-80b8-b0c290bedfc4.png">
<img width="764" alt="Screen Shot 2020-04-05 at 8 40 12 PM" src="https://user-images.githubusercontent.com/3240137/78514229-4b3b8d00-777e-11ea-8fc2-5ce17eada77c.png">
<img width="683" alt="Screen Shot 2020-04-05 at 8 40 20 PM" src="https://user-images.githubusercontent.com/3240137/78514230-4bd42380-777e-11ea-9ca9-f82215092f8c.png">

**Comment by parshap** [comment]
Thanks for the report! Do you want to send a PR to upgrade these dependencies?

**Comment by aorinevo** [comment]
Absolutely, I’ve already started.  Should have it ready for review in the next couple of days.

**Comment by aorinevo** [comment]
## Phase I:
- update `simple-git` from v1.104.0 to v1.132.0 (see https://github.com/steveukx/git-js/compare/v1.104.0...v1.132.0)
-- Github client (not sure why we aren't using octokit here)
- update `ora` from v2.1.0 to v4.0.3 (see https://github.com/sindresorhus/ora/compare/v2.1.0...v4.0.3)
- update `@types/ora` from v1.3.4 to v3.2.0 (see https://www.npmjs.com/package/@types/ora)
-- Note that ora provides its own type definition
-- Should we remove this then? (@parshap)
- update `joi`
-- installation via `joi` is deprecated as of v14.3.1 in favor of `@hapi/joi`
-- install `@hapi/joi` v
-- while a major version bump, joi is minimially used (see 
-- replace `@types/joi` with `@types/hapi__joi`
-- updated code and unit test to reflect delta in dep code

**Comment by aorinevo** [comment]
## Phase II
- update `js-yaml` from v3.12.0 to v3.13.1 (see https://github.com/nodeca/js-yaml/compare/3.12.0...3.13.1)
- update `chalk` from v2.4.1 to v4.0.0 (see https://github.com/chalk/chalk/compare/v2.4.1...v4.0.0)
- update `@types/js-yaml` from v3.12.1 to v3.12.3
- update `lodash` from v4.17.13 to v4.17.15 (see https://github.com/lodash/lodash/compare/4.17.13...4.17.15)
- update `log-symbols` from v2.2.0 to v3.3.0 (see https://github.com/sindresorhus/log-symbols/compare/v2.2.0...v3.0.0)

Notes
- tests pass
- tested locally on a couple of code migrations

**Comment by aorinevo** [comment]
## Phase III (see https://github.com/NerdWalletOSS/shepherd/pull/107):
- update `commander` from v2.18.0 to v5.0.0
- update `fs-extra` from v6.0.1 to v9.0.0
-- update `@types/fs-extra` from v5.0.4 to v8.1.0
- `fs-extra-promise`(already on latest)
-- update `@types/fs-extra-promise` from v1.0.7 to v1.0.8
- `netrc` (already on latest)

**Comment by aorinevo** [comment]
@parshap, PR #107 is ready for review.

**Comment by aorinevo** [comment]
<img width="878" alt="Screen Shot 2020-04-08 at 4 50 02 PM" src="https://user-images.githubusercontent.com/3240137/78832321-0c454b80-79b9-11ea-9954-6acc5b8d0a01.png">

**Comment by aorinevo** [comment]
This is is closed by way of #107

**Comment by parshap** [comment]
@aorinevo are the other prs you have open to upgrade dependencies ready for review as well?

**Comment by aorinevo** [comment]
I closed the other PRs as they were incremental steps towards PR #107 which closed Part I of the dependency management.

Next I’m going to tackle the jest dependencies but that will require some changes in the unit tests.  Should be ready by EOD tomorrow.

Note that once jest update is complete, vulnerabilities will drop to zero :)
